packages/core/src/internal/views/has-execution-failed.ts

```diff
@@ -20,6 +20,7 @@ export function hasExecutionFailed(
 
   return (
     exState.status === ExecutionStatus.FAILED ||
-    exState.status === ExecutionStatus.TIMEOUT
+    exState.status === ExecutionStatus.TIMEOUT ||
+    exState.status === ExecutionStatus.HELD
```

Comment:
Same

Reply:
This func is used to determine whether the next batch should be processed or whether execution should halt and the result be returned.

I have swapped this function with a `has-execution-succeeded` function for use in the batch processing logic. This means we can consider `Held` not an error, but still consider it as something worth halting processing for, without distorting the meaning of `hasExecutionFailed`.